service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -185,14 +185,32 @@ func (d *Decoder) decodeBinary(b []byte, v reflect.Value) error {
 		return nil
 	}
 
-	switch v.Interface().(type) {
-	case []byte:
+	if v.Kind() != reflect.Slice {
+		return &UnmarshalTypeError{Value: "binary", Type: v.Type()}
+	}
+
+	if v.Type() == byteSliceType {
```

Comment:
yeah in this case because `if v.Type() == byteSliceType {` only checks for a type being `[]byte` the following switch statement for at line [203](https://github.com/aws/aws-sdk-go/pull/747/files#diff-7b4de96174d722928e311dc8ec32d2fbR204) will capture types that are aliases of `[]byte`.